docs/electrical/embedded/styleguide.md

```diff
@@ -1,3 +1,136 @@
+5. ```volatile``` should be used in all cases where a variable can be modified by anything outside of the normal control flow of the code (i.e. an ISR).
+
+
+### Comment Rules
+1. Both single (//) and multi (/*) comment styles are considered acceptable, and should be used as applicable. When multi-line comments are used, all lines should begin with an asterisk (\*)
+2. Multi-line comments should maintain a level of indentation with the leading asterisks vertical to each other, as shown below.
+3. One leading space should be put between the comment text and the comment symbol.
+4. If placed at the end of a line of code, all comments should be vertically aligned.
+```c
+/* comment text
+ * comment text
+ * comment text
+ */
+
+ // comment text
+```
+
+
+5. All comments should be written in coherent, complete sentences.
+6. Assume a competentcy with the language from a reader, and avoid explaining the obvious. (*You aren't trying to hit a comment count for an assignment, you're trying to actually be helpful to yourself and others in the future*)
+7. Single line comments should be used where the function of a single line may be ambiguous.
+
+*TODO: Consider whether it's necessary to require conformance to automatic documentation generation standards*
+
+#### Comment Locations
+1. Functions should be preceeded by a short comment block explaining the relevant values (inputs, outputs) and an extremely brief description of the purpose of the function.
+```c title="Example"
+/* [sum]
+ * Returns the sum of two numbers
+ *
+ * Input: int A, int B
+ * Output: int C
+ */
+
+int sum(int A, int B) {
+    int c;
+    c = a + b;
+    return c;
+}
+```
+
+### White Space Rules
+1. The keywords ```if, while, for, switch,``` and ```return``` should be followed by a single space if there is more text on the same line.
+2. Unless specified by another whitespace rule, all operators should be surrounded on either side by a single space.
+3. Unary operators should not have a space between the operator and the operand.
+4. Structure pointer and member operators should have no whitespace on either side.
+5. Parentheses, braces, and brackets should not have an adjacent space on the interior side.
+6. Excepting the end of a line, separating commas should should be followed by one space.
+7. Semicolons should follow a preceeding statement without a space, but should always be followed by a space.
+8. Use whitespace to separate blocks of code from each other in a logical fashion. (William's note: to please my OCD use either 1 or 3 lines of separation depending on context, don't use two :) )
+
+#### Pointer (* and &) Operators
+1. When used with a type, such as in the case of ```pointerType name```, the pointer operator * should be appended to the type name (i.e. ```type* name```).
```

Comment:
NO! This creates bad habits. Consider the following

```c
int* a, b, c;
```

Many would assume all three (a, b, and c) are pointers. THEY ARE NOT! Only a is!

In contrast

```c
int *a, b, c;
```

Is easier for many people (especially with limited c experience) to identify that a is a pointer, but b and c are not.

Additionally, the rule as you propose it would not allow the following

```c
int *a, *b, *c;
```

Which is common and IMO improves readability of code in the right situations.

Reply:
"*" is a variable name modifier. NOT a type modifier. I understand that this itself is a point of confusion, however this is how the language works.

Reply:
Understood. That presents a further proposal, where pointers and non-pointers should be declared on separate lines to help minimize cases where one is accidentally declared as the other.